rlog.go

```diff
@@ -235,30 +250,143 @@ func (f filter) match(filename string, level int) (bool, bool) {
 	return false, false
 }
 
+// updateIfNeeded returns a new value for an existing config item. The priority
+// flag indicates whether the new value should always override the old value.
+// Otherwise, the new value will not be used in case the old value is already
+// set.
+func updateIfNeeded(oldVal string, newVal string, priority bool) string {
+	if priority || oldVal == "" {
+		return newVal
+	} else {
+		return oldVal
+	}
+}
+
+// updateConfigFromFile reads a configuration from the specified config file.
+// It merges the supplied config with the new values.
+func updateConfigFromFile(config *rlogConfig) {
+	settingConfFile = config.confFile
+	// If no config file was specified we will default to a known location.
+	if settingConfFile == "" {
+		execName := filepath.Base(os.Args[0])
+		settingConfFile = fmt.Sprintf("/etc/rlog/%s.conf", execName)
+	}
+
+	// Scan over the config file, line by line
+	file, err := os.Open(settingConfFile)
+	defer file.Close()
+	if err != nil {
+		// Any error while attempting to open the logfile ignored. In many
+		// cases there won't even be a config file.
+		return
+	}
+
+	scanner := bufio.NewScanner(file)
+	i := 0
+	for scanner.Scan() {
+		i++
+		line := strings.TrimSpace(scanner.Text())
+		if line == "" {
+			continue
+		}
+		tokens := strings.SplitN(line, "=", 2)
+		if len(tokens) == 0 {
+			continue
+		}
+		if len(tokens) != 2 {
+			fmt.Fprintf(os.Stderr, "WARN: rlog - Malformed line in config file %s:%d\n",
```

Comment:
I put the rlog-internal error reporting into a single function (rlogIssue), which currently writes to stderr, but which we can change later to do something different if we want. But at least we're not putting writes to stderr all over our code.